src/context/WalletProvider/WalletViewsSwitch.tsx

```diff
@@ -60,8 +60,9 @@ export const WalletViewsSwitch = () => {
   }
 
   const onContinue = useCallback(() => {
-    history.push('/select')
-  }, [history])
+    // Without this check we'll fire again once a KeepKey initializes and ask the user to select a wallet again
+    if (!initialRoute || initialRoute === '/') history.push('/select')
```

Comment:
This is bit hacky, though it works until we have time to look at the `WalletProvider` re-renders and subsequent side-effects.